packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -70,8 +70,19 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
         commands.registerCommand(VscodeCommands.DIFF, {
             isVisible: () => false,
             // tslint:disable-next-line: no-any
-            execute: async (left: URI, right: URI) => {
-                await this.diffService.openDiffEditor(new TheiaURI(left), new TheiaURI(right));
+            execute: async (left: URI, right: URI, label?: string) => {
+                if (!left || !right) {
+                    throw new Error(`${VscodeCommands.DIFF} command requires at least two URI arguments. Found left=${left}, right=${right} as arguments`);
+                }
+                if (!URI.isUri(left)) {
+                    throw new Error(`Invalid argument for ${VscodeCommands.DIFF.id} command with left argument. Expecting URI left type but found ${left}`)
+                }
+                if (!URI.isUri(right)) {
+                    throw new Error(`Invalid argument for ${VscodeCommands.DIFF.id} command with right argument. Expecting URI right type but found ${right}`)
+                }
+
+                const leftURI = new TheiaURI(left);
```

Comment:
doesn't it complain about unused `const`?

Reply:
it's used there https://github.com/theia-ide/theia/blob/ad0b31e9c6c69edbef4d324376d00b3b35b4bd9e/packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts#L85 no ?

Reply:
indeed :-)